### PR TITLE
spiders(kosovo): update the spider with the OCDS URL and logic

### PR DESCRIPTION
**Pull request by yolile**
ref CRM-7714#note-34

### CODE REVIEW

File: kingfisher_scrapy/spiders/kosovo.py [hunk 1]
```diff
@@ -8,37 +10,36 @@ class Kosovo(SimpleSpider):
...
-        for stage in stages:
-            yield self.build_request(
-                url.format(stage),
-                headers=headers,
-                formatter=components(-1)
-            )
+        # The API is slow even with short periods, so we request one day at a time.
+        delta = self.until_date - self.from_date
+        for days in reversed(range(delta.days + 1)):
+            start = self.from_date + timedelta(days=days-1)
```
Comment (jpmckinney) [inline kingfisher_scrapy/spiders/kosovo.py:33]: Can you check the API's logic? Typically I'd expect "from Nov 29 to 30" to mean "all of Nov 29 and all of Nov 30" (2 days), not "24:00:00 on Nov 29 to 23:59:00 on Nov 30" (1 day).

```suggestion
            start = self.from_date + timedelta(days=days - 1)
```

File: kingfisher_scrapy/spiders/kosovo.py [hunk 2]
```diff
@@ -8,37 +10,36 @@ class Kosovo(SimpleSpider):
...
+            end = self.from_date + timedelta(days=days)
+            url = f'https://ocdskrpp.rks-gov.net/krppapi/tenderrelease?endDateFrom=' \
```
Comment (jpmckinney) [inline kingfisher_scrapy/spiders/kosovo.py:35]: I would have expected flake8 to complain about an f-string without a placeholder.

```suggestion
            url = 'https://ocdskrpp.rks-gov.net/krppapi/tenderrelease?endDateFrom=' \
```

File: kingfisher_scrapy/spiders/kosovo.py (hunk 1 above)
Comment (yolile) [reply kingfisher_scrapy/spiders/kosovo.py:33]: Checked and, for example, https://ocdskrpp.rks-gov.net/krppapi/tenderrelease?endDateFrom=2021-11-26&endDateEnd=2021-11-27 only returns data from 2021-11-26 and similarly, https://ocdskrpp.rks-gov.net/krppapi/tenderrelease?endDateFrom=2021-11-25&endDateEnd=2021-11-26 only returns data from 2021-11-25 (the API filters is based on `tender/tenderPeriod/endDate`). So it is indeed just one day.

Comment (jpmckinney) [reply kingfisher_scrapy/spiders/kosovo.py]: Ok, please update the `until_date` docs to explain that this date is non-inclusive.

Comment (yolile) [reply kingfisher_scrapy/spiders/kosovo.py]: Done

**Review by jpmckinney — APPROVED** [review]

**Pull request merged**